v2/tools/generator/internal/functions/get_extended_resources_function.go

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package functions
+
+import (
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astbuilder"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/dave/dst"
+	"reflect"
+	"sort"
+)
+
+const (
+	ExtendedResourcesFunctionName = "GetExtendedResources"
+)
+
+type GetExtendedResourcesFunction struct {
+	idFactory astmodel.IdentifierFactory
+	resources []astmodel.TypeName
+}
+
+// Ensure GetExtendedResources properly implements Function
+var _ astmodel.Function = &GetExtendedResourcesFunction{}
+
+// NewGetExtendedResourcesFunction creates a new GetExtendedResources
+func NewGetExtendedResourcesFunction(idFactory astmodel.IdentifierFactory, resources []astmodel.TypeName) *GetExtendedResourcesFunction {
+
+	result := &GetExtendedResourcesFunction{
+		idFactory: idFactory,
+		resources: sortResources(resources),
+	}
+	return result
+}
+
+// getPackageRefs iterates through the resources and returns package references
+func getPackageRefs(resources []astmodel.TypeName) []astmodel.PackageReference {
+	var packageRefs []astmodel.PackageReference
+	// Package reference for return type
+	packageRefs = append(packageRefs, astmodel.KubernetesResourceType.PackageReference)
+
+	for _, typeDef := range resources {
+		packageRefs = append(packageRefs, typeDef.PackageReference)
+	}
+	return packageRefs
```

Comment:
Nitpit: I like to have blank lines between blocks and before return statements so the return statement is easier to spot.